run

```diff
@@ -68,6 +68,39 @@ if [ ! -z "$OPENDKIM_DOMAINS" ] ; then
   dkimConfig
   service opendkim start
 fi
+
+# Checking for user specified passwd file
+if [ ! -z "$SASL_Passwds" ] || [ -e "/etc/pam.d/smtp"] ; then
+  # Creating auth settings file
+  if [ ! -e /etc/postfix/sasl/smtpd.conf ] ; then
+    cat <<'EOF' > /etc/postfix/sasl/smtpd.conf
+pwcheck_method: saslauthd
+mech_list: CRAM-MD5 DIGEST-MD5 LOGIN PLAIN
+EOF
+  fi
+  
+  # Updating saslauthd config to work with chroot
+  sed -i 's|START=no|START=yes|' /etc/default/saslauthd
+  sed -i 's|OPTIONS="-c -m /var/run/saslauthd"|OPTIONS="-c -r -m /var/spool/postfix/var/run/saslauthd"|' /etc/default/saslauthd
+  dpkg-statoverride --add root sasl 710 /var/spool/postfix/var/run/saslauthd
+  
+  # Giving postfix permission to use saslauthd
+  adduser postfix sasl
+  
+  # Creating PAM authentication profile
+  if [ ! -e "/etc/pam.d/smtp" ] ; then
```

Comment:
Yes. In theory a user could volume mount over top of this to load user passwds from /etc/shadow (or any other PAM method for that matter) instead of from file like I/this change expect. I put this check here so that if someone wants to mess with it, the container doesn't overwrite the user's file. I expect that if someone is mounting this file, that they know more than me XD

Reply:
Ok! yeap agree